app/dashboard/components/Form.tsx

```diff
@@ -17,34 +18,48 @@ const Form: React.FC<Props> = ({populateData}: Props) => {
       setSelectedFile(files[0]);
     }
   };
-  
+
+  const roped = ["Sport", "Trad"];
+  const boulder = ["Boulder"]; //  nope let's do this w/o combinations too; looks like some of the boulder/trads were done as trad...filter again
+  const boulderStyleNaive = ["Flash", "Send", "Attempt"];
+  const ropedStyleNaive = ["Onsight", "Flash", "Redpoint", "Fell/Hung", "Pinkpoint"];
+
   const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-  
+
     if (selectedFile) {
       Papa.parse(selectedFile, {
         header: true,
         skipEmptyLines: true,
-        complete: (results:ParseResult<any>) => {
-          if( results.errors.length === 0) {
-            const resultsData:RawDataList = results.data 
-            // Could just process the data here, or we could use populateData to add it to the state of the Dashboard component and process the data there
-            populateData(resultsData)
+        complete: (results: ParseResult<any>) => {
+          if (results.errors.length === 0) {
+            const resultsData: RawDataList = results.data
+            // process here, Luke said
+            const ropedOnlyNaive = resultsData.filter((oneRoute) => roped.includes(oneRoute["Route Type"]));
+            ropedOnlyNaive.forEach(element => {
+              element.Rating = eliminateSlashesFromGrades(element.Rating);
+              element.Rating = flattenPlusAndMinusGrades(element.Rating);
+              element.Rating = removeRiskRating(element.Rating);
+              console.log(element.Rating)
+              // soundas: this seems a bit kludgy; TODO: how do I chain + assign concisely
+            });
+            const boulderOnlyNaive = resultsData.filter((oneRoute) => boulder.includes(oneRoute["Route Type"]) || boulderStyleNaive.includes(oneRoute["Style"]));
+            populateData(ropedOnlyNaive) // how...will populate now work. Luke, help! 
```

Comment:
I guess...I'm just going to pass ropedOnlyNaive to its own viz functions and boulderOnlyNaive to its own viz functions? 

@lukehtravis I need some structural oversight. help.

Reply:
You can recombine the arrays by doing

populateData([…ropedOnlyNaive, …boulderOnlyNaive])

The ellipse here is called the spread operator in javascript and it concisely joins arrays…we push all data into populateData, then can filter for bouldering data later on the vis side of things

https://www.w3schools.com/react/react_es6_spread.asp